code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,308 @@
+//Machine to hold a deployed gun. It aquired nearly all of its variables from the gun itself.
+/obj/machinery/deployable/mounted
+
+	anchored = TRUE
+	resistance_flags = XENO_DAMAGEABLE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	use_power = FALSE
+	max_integrity = 100
+	soft_armor = list("melee" = 0, "bullet" = 50, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 0, "fire" = 0, "acid" = 0)
+	hud_possible = list(MACHINE_HEALTH_HUD, SENTRY_AMMO_HUD)
+
+	///Gun that does almost all the work when deployed. This type should not exist with a null 'gun'. Typecasted version of internal_item
+	var/obj/item/weapon/gun/gun
+
+	///Icon state for when the gun has no ammo.
+	var/icon_empty
```

Comment:
should be generated automaticly

if(empty)
      icon_state = initial(icon_state)+"_empty" or whatever